### ISSUE TITLE
Fatal error in type inference with Pluto configuration

### ISSUE THREAD

**Issue by adrhill** [Roger-luo/Configurations.jl#88]
I'm currently running into issues with Configurations.jl and Pluto (CC @fonsp).

When using PlutoSliderServer, I get fatal type inference errors:
```Julia
julia> PlutoSliderServer.export_notebook("my_notebook.jl")
ERROR: fatal error in type inference (type bound)
Stacktrace:
  [1] foreach_keywords
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:146 [inlined]
  [2] collect_underscore_keywords!(list::Vector{Symbol}, ::Type{Pluto.Configuration.CompilerOptions}, prefix::Symbol)
    @ Configurations ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:180
...
```

I was able to narrow this down to Pluto's `CompilerOptions`

https://github.com/fonsp/Pluto.jl/blob/f3d8946dd2743e605418d73cead26e6fcd79d11f/src/Configuration.jl#L231-L255

which can be called with default arguments:
```Julia
julia> Pluto.Configuration.CompilerOptions()
Pluto.Configuration.CompilerOptions(nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, nothing, "no", "no", "4")
```

However, using `Configurations.from_kwargs` on this option appears to be the source of this error:
```Julia
julia> Configurations.from_kwargs(Pluto.Configuration.CompilerOptions)
ERROR: fatal error in type inference (type bound)
Stacktrace:
  [1] foreach_keywords
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:146 [inlined]
  [2] collect_underscore_keywords! (repeats 2 times)
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:180 [inlined]
  [3] underscore_keywords(#unused#::Type{Pluto.Configuration.CompilerOptions})
    @ Configurations ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:138
  [4] #from_underscore_kwargs!#25
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:57 [inlined]
  [5] from_underscore_kwargs! (repeats 2 times)
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:54 [inlined]
  [6] from_kwargs(convention!::typeof(Configurations.from_underscore_kwargs!), ::Type{Pluto.Configuration.CompilerOptions}; kw::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Configurations ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:14
  [7] from_kwargs
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:12 [inlined]
  [8] #from_underscore_kwargs#22
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:31 [inlined]
  [9] from_underscore_kwargs
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:30 [inlined]
 [10] #from_kwargs#21
    @ ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:23 [inlined]
 [11] from_kwargs(::Type{Pluto.Configuration.CompilerOptions})
    @ Configurations ~/.julia/packages/Configurations/uH65h/src/from_kwargs.jl:23
 [12] top-level scope
    @ REPL[52]:1
```
___
<details>
<summary>Dependencies</summary>

```
Status `~/Developer/julia-ml-course/Project.toml`
  [5218b696] Configurations v0.17.4
  [713c75ef] Franklin v0.10.79
  [c3e4b0f8] Pluto v0.19.22
  [2fc8631c] PlutoSliderServer v0.3.21
```

</details>

<details>
<summary>Version Info</summary>

```
Julia Version 1.9.0-rc1
Commit 3b2e0d8fbc1 (2023-03-07 07:51 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin21.4.0)
  CPU: 16 × Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, skylake)
  Threads: 4 on 8 virtual cores
Environment:
  JULIA_NUM_THREADS = 4
```

</details>

*Filed using [BrowserMacros.jl](https://github.com/adrhill/BrowserMacros.jl/)*

**Comment by Roger-luo** [comment]
It seems to be a regression of 1.9 then?

**Comment by adrhill** [comment]
Yes, I think so.

**Comment by Roger-luo** [comment]
It's probably better to cross post this issue at Julia repo then, I think I don't have the knowledge of fixing a type inference failure. Thanks!

**Comment by fonsp** [comment]
https://github.com/JuliaLang/julia/issues/49103

**Comment by fonsp** [comment]
Fixed in Julia 1.9.0-rc2
